components/post_view/post/post.jsx

```diff
@@ -165,14 +153,10 @@ class Post extends React.PureComponent {
         clearTimeout(this.highlightTimeout);
     }
 
-    componentDidUpdate(prevProps) {
+    componentDidUpdate() {
         if (this.state.a11yActive) {
             this.postRef.current.dispatchEvent(new Event(A11yCustomEventTypes.UPDATE));
         }
-        if (this.props.post.state === Posts.POST_DELETED && prevProps.post.state !== Posts.POST_DELETED) {
-            // ensure deleted message content does not remain in stale aria-label
-            this.updateAriaLabel();
-        }
```

Comment:
Is this one of the main culprits of the performance issue?

Reply:
Sort of. The main performance issue was because the return value of `createAriaLabelForPost` returns a new function whenever a bunch of different things change (notably, emojis when we load autocomplete results), so we'd end up re-rendering Post just because of a prop that we're not even using most of the time returns a new value.